src/views/Graph.vue

```diff
@@ -558,25 +956,66 @@ export default {
         `)
       }
 
-      if (this.groupCycle) {
-        // Loop over the subgraphs
-        Object.keys(cycles).forEach((key, i) => {
-          // Loop over the nodes that are included in the subraph
-          const nodeFormattedArray = cycles[key].map(a => `"${a.id}"`)
-          ret.push(`
-          subgraph cluster_margin_${i}
-          {
-            margin=100.0
-            label="margin"
-            subgraph cluster_${i} {${nodeFormattedArray};\n
-              label = "${key}";\n
-              fontsize = "70px"
-              style=dashed
-              margin=60.0
-            }
-          }`)
+      const graphSections = {}
+      Object.keys(cycles).forEach((cycle, iCycle) => {
+        const indexSearch = Object.values(this.cylcTree.$index).filter((node) => {
+          return node.tokens.cycle === cycle
         })
-      }
+        if (indexSearch.length && !this.collapseCycle.includes(cycle)) {
+          indexSearch.forEach((task) => {
+            const parent = task.node.firstParent
+            if (!parent) { return }
+            const section = graphSections[parent.id] ??= []
```

Comment:
Can avoid declaring an extra variable:

```suggestion
            if (!task.node.firstParent) { return }
            const section = graphSections[task.node.firstParent.id] ??= []
```